x-pack/legacy/plugins/index_management/public/components/mappings_editor/constants/field_options.tsx

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React from 'react';
+import { EuiText } from '@elastic/eui';
+
+import { DataType, ParameterName, SelectOption, SuperSelectOption, ComboBoxOption } from '../types';
+import { FIELD_OPTIONS_TEXTS, LANGUAGE_OPTIONS_TEXT, FieldOption } from './field_options_i18n';
+import { INDEX_DEFAULT, STANDARD } from './default_values';
+import { MAIN_DATA_TYPE_DEFINITION } from './data_types_definition';
+
+export const TYPE_ONLY_ALLOWED_AT_ROOT_LEVEL: DataType[] = ['join'];
+
+export const TYPE_NOT_ALLOWED_MULTIFIELD: DataType[] = [
+  ...TYPE_ONLY_ALLOWED_AT_ROOT_LEVEL,
+  'object',
+  'nested',
+  'alias',
+];
+
+export const FIELD_TYPES_OPTIONS = Object.entries(MAIN_DATA_TYPE_DEFINITION).map(
+  ([dataType, { label }]) => ({
+    value: dataType,
+    label,
+  })
+) as ComboBoxOption[];
+
+interface SuperSelectOptionConfig {
+  inputDisplay: string;
+  dropdownDisplay: JSX.Element;
+}
+
+export const getSuperSelectOption = (
+  title: string,
+  description: string
+): SuperSelectOptionConfig => ({
+  inputDisplay: title,
+  dropdownDisplay: (
+    <>
+      <strong>{title}</strong>
+      <EuiText size="s" color="subdued">
+        <p className="euiTextColor--subdued">{description}</p>
+      </EuiText>
+    </>
+  ),
+});
+
+const getOptionTexts = (option: FieldOption): SuperSelectOptionConfig =>
+  getSuperSelectOption(FIELD_OPTIONS_TEXTS[option].title, FIELD_OPTIONS_TEXTS[option].description);
+
+type ParametersOptions = ParameterName | 'languageAnalyzer';
+
+export const PARAMETERS_OPTIONS: {
```

Comment:
It doesn't look like this is being used. Can we remove it? If so, will this allow us to remove more code as well?